### DISCUSSION TITLE
Package updates not found when using custom regex manager

### DISCUSSION THREAD

**Discussion by sunker** [renovatebot/renovate#29315] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I'm using the [regex custom manager](https://docs.renovatebot.com/modules/manager/regex/) to update a certain devDependency in a non-standard package.json file. The file is called `template.json`, and it's partly filled with handlebar style templates. It looks something like this:

```json
{
  "name": "{{ kebabCase name }}",
  "version": "1.0.0",
  ...
  "devDependencies": {
    "@babel/core": "^7.21.4",
    "@types/react-router-dom": "^{{ reactRouterVersion }}",
    "@swc/helpers": "{{ version }}",
    "@example/plugin": "^1.0.1"
    ...
}
```

I'd like Renovate to update `@example/plugin` in `template.json`, but I cannot get it to work using the following config. There are new versions available for the `@example/plugin` dep, but no Renovate PRs are being raised.

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "includePaths": ["templates/"],
  "customManagers": [
    {
      "customType": "regex",
      "fileMatch": ["^template.json$"],
      "matchStrings": [
        "\"devDependencies\":\\s*\\{[^}]*}",
        "\"(?<depName>@example\/plugin)\"[^\"]*:\\s*\"(?<currentValue>.*)\""
      ],
      "datasourceTemplate": "npm"
    }
  ]
}
```
Any guidance would be very appreciated! 

### Logs (if relevant)

<details><summary>From the job output:</summary>

```
DEBUG: packageFiles with updates
{
  "baseBranch": "main"
  "config": {
    "regex": [
      {
        "deps": [
          {
            "depName": "@example/plugin",
            "currentValue": "^1.0.1",
            "datasource": "npm",
            "replaceString": "\"@example/plugin\": \"^1.0.1\"",
            "updates": [],
            "packageName": "@example/plugin",
            "versioning": "npm",
            "warnings": [],
            "sourceUrl": "https://github.com/example/repo",
            "registryUrl": "https://registry.npmjs.org",
            "sourceDirectory": "@example/plugin",
            "currentVersion": "1.2.2",
            "currentVersionTimestamp": "2024-05-28T11:47:27.553Z"
          }
        ],
        "matchStrings": [
          "\"devDependencies\":\\s*\\{[^}]*}",
          "\"(?<depName>@example/plugin)\"[^\"]*:\\s*\"(?<currentValue>.*)\""
        ],
        "datasourceTemplate": "npm",
        "packageFile": "packages/create-plugin/templates/common/_package.json"
      }
    ]
  }

```

</details>

**Comment by rarkins** [comment]
What update are you expecting? You are using a range so by default won't get any update unless the new version is newer than the range (e.g. v2.0.0)

**Reply by sunker** [reply]
Turns out I had a typo in my template. Sorry for this, and thanks @rarkins!
